server/server.go

```diff
@@ -127,113 +120,91 @@ func (s *Server) HandleEvent(ctx context.Context, e lookout.Event) error {
 	return err
 }
 
-// HandleReview sends request to analyzers concurrently
-func (s *Server) HandleReview(ctx context.Context, e *lookout.ReviewEvent, safePosting bool) error {
-	ctx, logger := ctxlog.WithLogFields(ctx, log.Fields{
-		"provider": e.Provider,
-	})
-	logger.Infof("processing pull request")
+func (s *Server) AnalyzeAndComment(ctx context.Context, e lookout.Event, safePosting bool) error {
+	comments, err := s.analyze(ctx, e, safePosting)
+	if err == nil {
+		if err := s.post(ctx, e, comments, safePosting); err != nil {
+			s.status(ctx, e, lookout.ErrorAnalysisStatus)
+			err = fmt.Errorf("posting analysis failed: %s", err)
+		} else {
+			s.status(ctx, e, lookout.SuccessAnalysisStatus)
+		}
+	}
+	return err
+}
 
+func (s *Server) analyze(ctx context.Context, e lookout.Event, safePosting bool) ([]lookout.AnalyzerComments, error) {
+	ctxlog.Get(ctx).Infof("processing event type %d", e.Type())
```

Comment:
I think we're also missing info about the event provider (see https://github.com/src-d/lookout/pull/443/commits/13266785477a3562816e9a9bcb19f38136552103#diff-91bbeda7eb98a7adc57b9e47e2cf5c2bL188)

Reply:
You're right, I forgot to add it in to the log ctx.